hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/RunCompactionActionExecutor.java

```diff
@@ -85,9 +86,12 @@ public HoodieWriteMetadata<HoodieData<WriteStatus>> execute() {
 
       // try to load internalSchema to support schema Evolution
       HoodieWriteConfig configCopy = config;
-      Pair<Option<String>, Option<String>> schemaPair = InternalSchemaCache
-          .getInternalSchemaAndAvroSchemaForClusteringAndCompaction(table.getMetaClient(), instantTime);
-      if (schemaPair.getLeft().isPresent() && schemaPair.getRight().isPresent()) {
+      boolean schemaEvolutionEnable = new TableSchemaResolver(table.getMetaClient()).getTableInternalSchemaFromCommitMetadata().isPresent();
+      Pair<Option<String>, Option<String>> schemaPair = Pair.of(Option.empty(), Option.empty());
+      if (schemaEvolutionEnable) {
+        schemaPair = InternalSchemaCache.getInternalSchemaAndAvroSchemaForClusteringAndCompaction(table.getMetaClient(), instantTime);
```

Comment:
Optimize the code, trigger the corresponding logic only when the schema evolution is enabled